test/realm_value_test.dart

```diff
@@ -0,0 +1,158 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+
+import 'package:test/test.dart' hide test, throws;
+import '../lib/realm.dart';
+
+import 'test.dart';
+
+part 'realm_value_test.g.dart';
+
+@RealmModel()
+class _AnythingGoes {
+  @Indexed()
+  late RealmValue oneAny;
+  late List<RealmValue> manyAny;
+}
+
+void main() {
```

Comment:
Will be good to make more tests for example:
- Query by RealmValue;
- Frozen objects with RealmValue;
- Embedded objects with RealmValue;
- Flexible sync with RealmValue;
- Subscriptions by RealmValue;

Reply:
This would test realm-core more than realm-dart.

Reply:
Some of these have the potential to surface symmetric bugs and may be valuable. Let's say you make a mistake and you store double values as negative the original value (e.g. `4.56 -> RealmValue(-4.56)`). This kind of bug will always pass a test like:

```
final value = 4.56;
realm.write(() {
  myObj.any = value;
});

expect(myObj.any, value);
```

Ideally, that should be tested by serializing the object to json and comparing the stringified representation as seen by Core, but those other tests may also be valuable. While I agree it's testing Core, historically, SDKs have been the primary source of bug reports for the Core team and it doesn't hurt to have a few extra tests.

Reply:
As the 6th SDK I see less value in tests that essentially test realm-core. Adding them now will delay this PR, and require maintenance down the line.

Reply:
.. an argument can be made that we are only the 2nd SDK to use the c-api though